[lutfir-cpu] iP

lutfir-cpu opened the pull request:
# LutChat

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

LutChat frees your mind of having to remember things you need to do. It's,
- **text-based**
- **easy to learn**
- **~~FAST~~ SUPER FAST** to use

All you need to do is,
1. [Download it from here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you. 😌

And it is **FREE**!

## Features:
- [x] Managing tasks  
- [ ] Managing deadlines (coming soon)  
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

lutfir-cpu committed:
Changed name of chatbot, added greet and exit

lutfir-cpu committed:
added echo

lutfir-cpu committed:
added 'add' and 'list' functionality

lutfir-cpu committed:
added mark and unmark for tasks

lutfir-cpu committed:
added todo, deadline and event, added testing

lutfir-cpu committed:
fixed EXPECTED.txt

lutfir-cpu committed:
added error handling and delete functionality

lutfir-cpu committed:
making sure code is pushed to the correct branch

lutfir-cpu committed:
Revert "making sure code is pushed to the correct branch"

This reverts commit b0307231fc0795053cd7c167061db6eb16801bd1.

lutfir-cpu committed:
checking if commit is on the right branch

lutfir-cpu committed:
test: comment should be in correct branch

lutfir-cpu committed:
Added loadTasks and saveTasks features

lutfir-cpu committed:
.

lutfir-cpu committed:
Merge branch 'master' into branch-Level-7

lutfir-cpu committed:
Add Date feature

lutfir-cpu committed:
Merge branch 'branch-Level-8'

lutfir-cpu committed:
Refactor classes to fit OOP

lutfir-cpu committed:
Merge branch 'A-MoreOOP'

lutfir-cpu committed:
Revert "Merge branch 'A-MoreOOP'"

This reverts commit 9439fb96ef7b101164121f384a8bda807c3c30e6, reversing
changes made to 5818a5ac9a1ecc6ae0d2b519f41bd583aea74480.

lutfir-cpu committed:
Merge branch 'A-MoreOOP'

lutfir-cpu committed:
Refactor classes to fit OOP

lutfir-cpu committed:
Merge branch 'A-MoreOOP'

lutfir-cpu committed:
Merge branch 'master' of https://github.com/lutfir-cpu/ip

lutfir-cpu committed:
Revert "Merge branch 'branch-Level-8'"

This reverts commit 5818a5ac9a1ecc6ae0d2b519f41bd583aea74480, reversing
changes made to 83ace0bc40042f4d2ca5457f46e4e8aef10d9927.

lutfir-cpu committed:
Test Commit

lutfir-cpu committed:
Test commit

lutfir-cpu committed:
Refactor code to fit OOP

lutfir-cpu committed:
Refactor code into lutchat package

lutfir-cpu committed:
Merge branch 'A-Packages'

lutfir-cpu committed:
Merge branch 'add-gradle-support' of https://github.com/lutfir-cpu/ip

lutfir-cpu committed:
Configure 'Run' command in build.gradle

lutfir-cpu committed:
Merge tag 'A-Gradle'

lutfir-cpu committed:
Add TaskListTest

lutfir-cpu committed:
Add EventTest

lutfir-cpu committed:
Merge branch 'A-JUnit'

Incogdino commented on src/main/java/lutchat/Parser.java:
```diff
@@ -0,0 +1,96 @@
+        switch (keyword) {
+            case "bye":
+                return false;
+
```
Switch case does not follow the coding standard. There should not be an indent for the switch case.

Incogdino commented on src/main/java/lutchat/Storage.java:
```diff
@@ -0,0 +1,87 @@
+            String desc = parts[2];
+
+            Task task = null;
+            switch (taskType) {
```
Switch case does not follow the coding standard. There should not be an indent for the switch case.

Incogdino commented on src/main/java/lutchat/Storage.java:
```diff
@@ -0,0 +1,87 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
```
JavaDoc comments can be added to help improve code readability?

Incogdino commented on src/main/java/lutchat/Task.java:
```diff
@@ -0,0 +1,41 @@
+    public String toString() {
+        if (done) {
+            return "[X] " + desc;
+        } else
```
`else` construct is missing brackets {}. This is not inline with the java coding standard

Incogdino commented on src/main/java/lutchat/Todo.java:
```diff
@@ -0,0 +1,22 @@
+package lutchat;
+
+public class Todo extends Task{
```
There is a missing whitespace here, it should be,
```suggestion
public class Todo extends Task {
```

Incogdino reviewed (commented):
LGTM. Just a few nits to fix..

lutfir-cpu committed:
Add Javadoc comments for Lutchat.java

lutfir-cpu committed:
Add Javadoc Comments to all classes

lutfir-cpu committed:
Modify to adhere to coding standards

lutfir-cpu committed:
Add Find feature

lutfir-cpu committed:
Merge branch 'branch-A-CodingStandard'

lutfir-cpu committed:
Merge branch 'branch-A-Javadoc'

# Conflicts: (resolved)
#	src/main/java/lutchat/Task.java
#	src/main/java/lutchat/Todo.java

lutfir-cpu committed:
Merge branch 'branch-Level-9'

# Conflicts: (resolved)
#	src/main/java/lutchat/Parser.java
#	src/main/java/lutchat/Task.java

lutfir-cpu committed:
Update to match Coding Standard

Incogdino commented on src/main/java/lutchat/TaskList.java:
```diff
@@ -0,0 +1,107 @@
+ * The TaskList class manages a list of tasks.
+ * It provides methods to add, delete, mark as done, and unmark tasks in the list.
+ */
+public class TaskList {
```
Good use of nouns for things and verbs for actions in your fields and methods

Incogdino commented on src/main/java/lutchat/Task.java:
```diff
@@ -0,0 +1,96 @@
+     *
+     * @return A string representing additional task details for file storage.
+     */
+    public abstract String additionalDescDetailsToFileFormat();
```
I like the good use of names to describe the method. It helps to keep to explain the named entity to the reader accurately and at a sufficient level of detail.

Incogdino commented on src/main/java/lutchat/Task.java:
```diff
@@ -0,0 +1,96 @@
+ * The Task class represents a task in the task list.
+ * It serves as an abstract base class for different types of tasks such as Todo, Deadline, and Event.
+ */
+public abstract class Task {
```
Great use of abstraction to not allow `Task` class to be instantiated

Incogdino commented on src/main/java/lutchat/Storage.java:
```diff
@@ -0,0 +1,107 @@
+     * @return An ArrayList of tasks loaded from the file.
+     * @throws IOException If an I/O error occurs while reading the file.
+     */
+    public ArrayList<Task> load() throws IOException {
```
I feel maybe this method could be named with more description like `loadTasks`

Incogdino commented on src/main/java/lutchat/Storage.java:
```diff
@@ -0,0 +1,107 @@
+     *
+     * @param taskList The TaskList object containing the tasks to be saved.
+     */
+    public void save(TaskList taskList) {
```
I feel maybe this method could be named with more description like `saveTasks`

Incogdino reviewed (commented):
LGTM, the code is easy to read. Methods and classes are well named. 

This is a follow up review as I misread that the second review was on Naming and not code quality standard

lutfir-cpu committed:
Added JavaFX Skeleton, haven't configured with lutchat

Runus01 commented on src/main/java/lutchat/Storage.java:
```diff
@@ -0,0 +1,107 @@
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
```
You could add a blank line here to separate the java.io imports from the java.util imports according to the coding conventions. It would also make the import packages more readable if they were in alphabetical order.

Runus01 commented on src/main/java/lutchat/Task.java:
```diff
@@ -0,0 +1,96 @@
+    /**
+     * Marks the task as not done.
+     */
+    public void markAsUndone() {
```
I used the term `markAsNotDone()` which may be a bit simpler but it's just a personal preference.

Runus01 commented on src/main/java/lutchat/Ui.java:
```diff
@@ -0,0 +1,125 @@
+ * The Ui class handles the user interface interactions for the application.
+ * It provides methods to display messages and prompt the user for input.
+ */
+public class Ui {
```
I like that you have a dedicated `Ui` class for handling output from the chatbot. I left the handling of output to my main method which does make things a bit messy in Duke.java

Runus01 commented on src/main/java/lutchat/Task.java:
```diff
@@ -0,0 +1,96 @@
+     *
+     * @return A string representing the task in a format suitable for file storage.
+     */
+    public String toFileFormat() {
```
Great idea to have Task.java handle the conversion of a Task object `toFileFormat()`. This abstraction would make it simpler to save the tasks to the data file when needed.

Runus01 commented on src/main/java/lutchat/Task.java:
```diff
@@ -0,0 +1,96 @@
+     */
+    @Override
+    public String toString() {
+        return (done ? "[X] " : "[ ] ") + description;
```
Great job in using the short-hand if / else statement here. It makes the method much more readable.

lutfir-cpu committed:
- Configured GUI to Lutchat
- Fixed Parser.java

lutfir-cpu committed:
Merge branch 'branch-Level-10'

lutfir-cpu committed:
Add configured Jar functionality

lutfir-cpu committed:
Merge branch 'branch-A-Jar'

lutfir-cpu committed:
Add assertions in Parser.java

lutfir-cpu committed:
Add assertions in Storage.java

lutfir-cpu committed:
Refactor Parser.java

lutfir-cpu committed:
Refactor Storage.java

lutfir-cpu committed:
Merge pull request #1 from lutfir-cpu/branch-A-Assertions

Merge branch-A-Assertions

lutfir-cpu committed:
Merge branch 'master' into branch-A-CodeQuality

lutfir-cpu committed:
Merge pull request #2 from lutfir-cpu/branch-A-CodeQuality

Merge branch-A-CodeQuality

lutfir-cpu committed:
Update README.md

lutfir-cpu committed:
Enhance search functionality in Parser::handleFindTask()

- Updated the search function in `handleFindTask` to use substring matching, allowing users to find tasks with partial matches.
- Implemented keyword extraction logic to improve search accuracy based on the user's input, trimming unnecessary spaces.
- The search now uses the `contains` method to locate tasks, improving the flexibility and user experience when finding tasks with non-exact matches.
- Overall improvements ensure the search feature is more user-friendly and intuitive, enhancing task management within Lutchat.

This commit improves the chatbot's search functionality, allowing for more robust and flexible task lookups, and adds error messaging for incomplete queries.

lutfir-cpu committed:
Merge branch 'branch-C-BettterSearch'

lutfir-cpu committed:
Merge branch 'master' of https://github.com/lutfir-cpu/ip

lutfir-cpu committed:
Update README.md

lutfir-cpu committed:
Refactor 'done' to 'isDone' to improve code quality and naming conventions

- Renamed the 'done' variable to 'isDone' in Task class to follow standard Java naming conventions for boolean variables.
- Updated all occurrences of 'done' to 'isDone' throughout the codebase, ensuring consistency and clarity.
- Improved readability by aligning with the convention of using "is" as a prefix for boolean values, making it immediately clear that the variable represents a boolean state.
- Refactored unit tests to reflect the new variable name and ensured all tests pass after the change.

This change enhances code maintainability and readability, improving overall code quality by following industry best practices.

lutfir-cpu committed:
Add product screenshot

lutfir-cpu committed:
Update README.md

lutfir-cpu committed:
Update README.md

lutfir-cpu committed:
Update README.md

lutfir-cpu committed:
Update README.md

lutfir-cpu committed:
Update README.md

lutfir-cpu committed:
Improve SLAP, fix User Guide

lutfir-cpu committed:
Update User Guide

lutfir-cpu committed:
Update javadoc commenst for Parser class
